core/service/cleaner.go

```diff
@@ -24,10 +24,12 @@ type Cleaner struct {
 
 // SessionStorage keeps sessions and allows removing them by proposal id
 type SessionStorage interface {
-	RemoveForService(serviceId string)
+	RemoveForService(serviceID string)
 }
 
 // Cleanup removes sessions of stopped service
-func (cleaner *Cleaner) Cleanup(instance *Instance) {
-	cleaner.SessionStorage.RemoveForService(string(instance.id))
+func (cleaner *Cleaner) Cleanup(event EventPayload) {
```

Comment:
Its probably not a `Cleanup` then, since imho its should be called only once and we don't care about event's state.